week-1-ts-ecosystem/w01e05-my-own-cli/src/showcase-demo.ts

```diff
@@ -0,0 +1,184 @@
+import { program } from 'commander'
+import {
+  ActionTypes,
+  CartType,
+  chooseAction,
+} from './showcaseActions/chooseAction'
+import { chooseCart } from './showcaseActions/chooseCart'
+import {
+  AuctionProduct,
+  BuyNowProduct,
+  FreeProduct,
+} from './shared_w01e03/Products'
+import Table from 'cli-table'
+import { removeItems } from './showcaseActions/removeItems'
+import { addItems } from './showcaseActions/addItems'
+import { faker } from '@faker-js/faker/locale/pl'
+import { logger } from './utils'
+import { DataRepository } from './DataRepository'
+import { PersistantCart } from './PersistantCart'
+
+program
+  .name('demo-showcase')
+  .description('CLI for demo showcase')
+  .version('0.1.0')
+program.option('--persistence', 'saves data to json file')
+
+program.parse()
+
+const options: { persistence: boolean } = program.opts() || {
+  persistence: false,
+}
+
+logger.info(`Persistence mode is: ${options.persistence}`)
+
+const AuctionsDataRepository = new DataRepository<AuctionProduct>(
+  'auctions',
+  options.persistence
+)
+const BuyNowDataRepository = new DataRepository<BuyNowProduct>(
+  'buyNow',
+  options.persistence
+)
+const FreeCartRepository = new DataRepository<FreeProduct>(
+  'forFree',
+  options.persistence
+)
+
+const AuctionCart = new PersistantCart<AuctionProduct>(AuctionsDataRepository)
+const BuyNowCart = new PersistantCart<BuyNowProduct>(BuyNowDataRepository)
+const FreeCart = new PersistantCart<FreeProduct>(FreeCartRepository)
```

Comment:
te stałe z małej powinny być `camelCase` np. `auctionCart` - bo tak to sugeruje że to klasy.